context/app/api/client.py

```diff
@@ -65,18 +62,34 @@ def get_all_dataset_uuids(self):
             raise Exception('At least 10k datasets: need to make multiple requests')
         return uuids
 
-    def get_entity(self, uuid=None, hbm_id=None):
-        if self.is_mock:
-            return {
-                'created': '2020-01-01 00:00:00',
-                'modified': '2020-01-01 00:00:00',
-                'provenance_user_displayname': 'Chuck McCallum',
-                'provenance_user_email': 'mccalluc@example.com',
-                'provenance_group_name': 'Mock Group',
-                'hubmap_id': 'abcd-1234',
-                'description': 'Mock Entity'
-            }
+    def get_all_donors(self, non_metadata_fields):
+        return self._get_all_entities_of_type(non_metadata_fields, 'Donor')
+
+    def get_all_samples(self, non_metadata_fields):
+        return self._get_all_entities_of_type(non_metadata_fields, 'Sample')
+
+    def get_all_datasets(self, non_metadata_fields):
+        return self._get_all_entities_of_type(non_metadata_fields, 'Dataset')
 
+    def _get_all_entities_of_type(self, non_metadata_fields, entity_type):
+        size = 10000  # Default ES limit
+        query = {
+            "size": size,
+            "post_filter": {
+                "term": {"entity_type.keyword": entity_type}
+            },
+            "_source": [*non_metadata_fields, 'mapped_metadata', 'metadata']
+        }
+        response_json = self._request(
+            current_app.config['ELASTICSEARCH_ENDPOINT']
+            + current_app.config['PORTAL_INDEX_PATH'],
+            body_json=query)
+        sources = [hit['_source'] for hit in response_json['hits']['hits']]
+        if len(sources) == size:
+            raise Exception('At least 10k datasets: need to make multiple requests')
```

Comment:
If we're not enabling the means to change the result window couldn't we just use `hits.hits.total.value` and if it's greater than 10,000 throw the exception?